library/src/main/java/com/android/mms/service_alt/MmsConfig.java

```diff
@@ -559,13 +560,14 @@ private static String getLine1(Context context, int subId) {
                     Context.TELEPHONY_SERVICE);
 
             if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP_MR1) {
-                return telephonyManager.getLine1Number();
+                return Utils.getMyPhoneNumber(context);
             } else {
                 try {
                     Method method = telephonyManager.getClass().getMethod("getLine1NumberForSubscriber", int.class);
-                    return (String) method.invoke(telephonyManager, subId);
+                    String number = (String) method.invoke(telephonyManager, subId);
+                    return number;
```

Comment:
This was the only part I was a little sketchy on. In this implementation, the lib will still call `getLine1NumberForSubscriber` rather than `Utils.getMyPhoneNumber`

How do you think we should handle this?

Reply:
thanks Luke! will look to find some time this weekend to finish it off 🙂